src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
+    private static final long TIMEOUT = 1000;
+    private static final String USER_DETAILS = "Details for %s";
+
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() ->
+           cache.computeIfAbsent(
+                   userId, this::getUserData), executor
+        );
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            TimeUnit.MILLISECONDS.sleep(TIMEOUT);
+            return new UserData(userId, USER_DETAILS.formatted(userId));
```

Comment:
Be cautious with using String.formatted() as it was introduced in Java 13. If you're working on a project that needs to be compatible with earlier versions of Java, consider using String.format() instead.